src/sqlparser/src/parser.rs

```diff
@@ -3550,28 +3552,20 @@ impl Parser<'_> {
     }
 
     pub fn parse_function_definition(&mut self) -> PResult<FunctionDefinition> {
-        let peek_token = self.peek_token();
-        match peek_token.token {
-            Token::DollarQuotedString(value) => {
-                self.next_token();
-                Ok(FunctionDefinition::DoubleDollarDef(value.value))
-            }
-            _ => Ok(FunctionDefinition::SingleQuotedDef(
-                self.parse_literal_string()?,
-            )),
-        }
+        alt((
+            single_quoted_string.map(FunctionDefinition::SingleQuotedDef),
+            dollar_quoted_string.map(FunctionDefinition::DoubleDollarDef),
+            Self::parse_identifier.map(|i| FunctionDefinition::Identifier(i.value)),
```

Comment:
When I said _(unquoted or double quoted) **identifier**_ I was thinking of `.real_value()` but now I realize `.value` has better backward compatibility:

When the SQL to parse is `as camelCase`, the old behavior gets `camelCase` but the identifier's `real_value` is `camelcase`. Although `as "camelCase"` is a case-sensitive identifier, I agree it is better to encourage `as 'camelCase'` if breaking change is inevitable.